lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -684,6 +684,13 @@ class DecomposeAtenSelectIntOp : public OpRewritePattern<AtenSelectIntOp> {
                              /*keepDim=*/true),
         op.getSelf(), dim, start, startPlusOne, /*step=*/one);
 
+    auto sliceTy = cast<BaseTensorType>(slice.getType());
+    auto resultTy = cast<BaseTensorType>(op.getResult().getType());
+    if (sliceTy.getSizes().size() == resultTy.getSizes().size()) {
```

Comment:
Okay, so you are checking if the slice should be squeezed or not, correct? Is this sort of like double protecting against the no-op squeezes, since the folder above should get rid of them in that case anyway?

Reply:
Correct - for some reason we were generating an illegal squeeze (it came up during the folder). I corrected this so we would no longer generate an illegal squeeze op.